ggshield/core/git_shell.py

```diff
@@ -22,6 +23,19 @@
 logger = logging.getLogger(__name__)
 
 
+class Filemode(Enum):
```

Comment:
`utils` is dependant on `git_shell`.
I wanted to avoid circular imports or local imports in the function.
Also, I think it makes sense to have git Filemodes in `git_shell`. Maybe I could make a `git_utils` file instead, WDYT ?

Reply:
No, the actual is fine then

Reply:
+1 for this. The less code we have in core.utils, the better.